frontend/apps/reader/modules/readeruserhyph.lua

```diff
@@ -0,0 +1,248 @@
+    USER_DICT_ERROR_NOT_SORTED = 3,
+}
+
+-- returns path to the user dictionary
+function ReaderUserHyph:getDictionaryPath()
+    return FFIUtil.joinPath(DataStorage:getSettingsDir(),
+        "user-" .. tostring(self.ui.document:getTextMainLangDefaultHyphDictionary():gsub(".pattern$", "")) .. ".hyph")
+end
+
+-- Load the user dictionary suitable for the actual language
+-- if reload==true, force a reload
+-- Unload is done automatically when a new dictionary is loaded.
+function ReaderUserHyph:loadDictionary(name, reload)
+    if G_reader_settings:isTrue("hyph_user_dict") and lfs.attributes(name, "mode") == "file" then
+        local ret = self.ui.document:setUserHyphenationDict(name, reload)
+        -- this should only happen, if a user edits a dictionary by hand or the user messed
+        -- with the dictionary file by hand. -> Warning and disable.
+        if ret == self.USER_DICT_ERROR_NOT_SORTED then
+            UIManager:show(InfoMessage:new{
+                text = T(_("The user dictionary\n%1\nis not alphabetically sorted.\n\nIt will not be used."), name),
+            })
+            logger.warn("UserHyph: Dictionary " .. name .. " is not sorted alphabetically.")
+            G_reader_settings:makeFalse("hyph_user_dict")
+        elseif ret == self.USER_DICT_MALFORMED then
+            UIManager:show(InfoMessage:new{
+                text = T(_("The user dictionary\n%1\nhas corrupted entries.\n\nOnly valid entries will be used."), name),
+            })
+            logger.warn("UserHyph: Dictionary " .. name .. " has corrupted entries.")
+        end
+    else
+        self.ui.document:setUserHyphenationDict() -- clear crengine user hyph dict
+    end
+end
+
+-- Reload on change of the hyphenation language
+function ReaderUserHyph:onTypographyLanguageChanged()
+    self:loadUserDictionary()
+end
+
+-- Reload on "ChangedUserDictionary" event,
+-- doesn't load dictionary if filesize and filename haven't changed
+-- if reload==true reload
+function ReaderUserHyph:loadUserDictionary(reload)
+    self:loadDictionary(self:isAvailable() and self:getDictionaryPath() or "", reload and true or false)
+    self.ui:handleEvent(Event:new("UpdatePos"))
+end
+
+-- Functions to use with the UI
+
+function ReaderUserHyph:isAvailable()
+    return G_reader_settings:isTrue("hyph_user_dict") and self:_enabled()
+end
+
+function ReaderUserHyph:_enabled()
+    return self.ui.typography.hyphenation
+end
+
+-- add Menu entry
+function ReaderUserHyph:getMenuEntry()
+    return {
+        text = _("Additional user dictionary"),
+        help_text = _([[The hyphenation rules for individual words can be changed with the user dictionary.
+Words not in that dictionary will be hyphenated by the selected method.
+
+Hyphenation can be changed by long pressing more than three seconds on a word and selecting 'Hyphenate' in the popup menu.
+
+If a word is removed from the user dictionary, the selected hyphenation method is applied again.]]),
```

Comment:
A bit verbose and redundant. This says as much, but feels too short :):
The hyphenation of a word can be changed from its default by long pressing for 3 seconds and selecting 'Hyphenate'.
I feel we shouldn't use the words "dictionary" or "method" in there, which are a bit implementation wordings.